sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/Jsr305AnnotationsCheck.java

```diff
@@ -0,0 +1,1096 @@
+            "jsr305.overridden.definitions.with.increased.param.constraint";
+
+    /** Key for error message. */
+    public static final String MSG_REDUNDANT_NONNULL_PARAM_ANNOTATION =
+            "jsr305.redundant.nonnull.param.annotation";
+
+    /** Key for error message. */
+    public static final String MSG_REDUNDANT_NULLABLE_PARAM_ANNOTATION =
+            "jsr305.redundant.nullable.param.annotation";
+
+    /** Key for error message. */
+    public static final String MSG_PARAMETER_WITHOUT_NULLNESS_ANNOTATION =
+            "jsr305.parameter.without.nullness.annotation";
+
+    /** Key for error message. */
+    public static final String MSG_RETURN_VALUE_WITH_NONNULL_BY_DEFAULT =
+            "jsr305.return.value.with.nonnull.by.default.annotation";
+
+    /** Key for error message. */
+    public static final String MSG_RETURN_VALUE_WITH_NULLABLE =
+            "jsr305.return.value.with.nullable.annotation";
+
+    /** Key for error message. */
+    public static final String MSG_CONTRADICTING_RETURN_VALUE_ANNOTATIONS =
+            "jsr305.contradicting.return.value.annotations";
+
+    /** Key for error message. */
+    public static final String MSG_OVERRIDDEN_METHOD_WITH_CHECK_RETURN_VALUE =
+            "jsr305.overridden.method.with.check.return.value.annotation";
+
+    /** Key for error message. */
+    public static final String MSG_REDUNDANT_NONNULL_BY_DEFAULT_ANNOTATION =
+            "jsr305.redundant.nonnull.by.default.annotation";
+
+    /** Key for error message. */
+    public static final String MSG_REDUNDANT_NULLABLE_BY_DEFAULT_ANNOTATION =
+            "jsr305.redundant.nullable.by.default.annotation";
+
+    /** Key for error message. */
+    public static final String MSG_VOID_WITH_CHECK_RETURN_VALUE_ANNOTATION =
+            "jsr305.void.with.check.return.value.annotation";
+
+    /** Key for error message. */
+    public static final String MSG_REDUNDANT_NONNULL_RETURN_ANNOTATION =
+            "jsr305.redundant.nonnull.return.annotation";
+
+    /** Key for error message. */
+    public static final String MSG_RETURN_WITHOUT_NULLNESS_ANNOTATION =
+            "jsr305.return.without.nullness.annotation";
+
+    /** Key for error message. */
+    public static final String MSG_OVERRIDDEN_METHODS_ALLOW_ONLY_NONNULL =
+            "jsr305.overridden.methods.allow.only.nonnull";
+
+    /** Key for error message. */
+    public static final String MSG_NEED_TO_INHERIT_PARAM_ANNOTATIONS =
+            "jsr305.need.to.inherit.param.annotations";
+
+    /** Key for error message. */
+    public static final String MSG_CONSTRUCTOR_WITH_RETURN_ANNOTATION =
+            "jsr305.constructor.with.return.annotation";
+
+    /** Package name. */
+    private static final String PKG_JAVAX_ANNOTATION = "javax.annotation";
+
+    /**
+     * Class NullnessAnnotation. The annotations we consider as "nullness-relevant".
+     */
+    protected enum NullnessAnnotation {
```

Comment:
please make it private.
We highly discourage hierarchy between Checks.

Reply:
Done.